spring-orm/src/main/java/org/springframework/orm/jpa/vendor/HibernateJpaDialect.java

```diff
@@ -118,6 +121,34 @@ public void setPrepareConnection(boolean prepareConnection) {
 		this.prepareConnection = prepareConnection;
 	}
 
+	/**
+	 * Set, whether to release connection after transaction is commited or rolled
+	 * back. Only works for pre-bound sessions.
+	 * <p> This setting is needed to work around the fact, that it is not possible
+	 * to use {@link org.hibernate.resource.jdbc.spi.PhysicalConnectionHandlingMode#DELAYED_ACQUISITION_AND_RELEASE_AFTER_TRANSACTION}
+	 * handling mode without sacrificing setting non-default isolation levels
+	 * or read-only flag. Releasing connection might prevent connection pool
+	 * starvation if open-in-view is on.
+	 * <p> Default is "false" for backward compatibility. If you turn this flag
+	 * on it still does nothing unless session is pre-bound (most likely if
+	 * open-in-view) is off. If session is pre-bound and the flag is on, then
+	 * after transaction is finished (successfully or not), underlying JDBC
+	 * connection will be released and acquired later according to {@link org.hibernate.resource.jdbc.spi.PhysicalConnectionHandlingMode}
+	 * (is set to {@link org.hibernate.resource.jdbc.spi.PhysicalConnectionHandlingMode#DELAYED_ACQUISITION_AND_HOLD}
+	 * with {@link HibernateJpaVendorAdapter#getJpaPropertyMap()} if you don't
+	 * specify it yourself).
+	 * <p> Please pay attention, that this setting doesn't affect how Hibernate
+	 * handles connections, which were acquired on-demand, to lazily load
+	 * collections outside of transaction context.
+	 * <p> Specifically, connections, acquired to serialize entities, returned
```

Comment:
I do not understand this sentence and its purpose. What connections "acquired to serialize entities" are you talking about here?

> Specifically, connections, acquired to serialize entities, returned by rest controller method will only be closed, after serialization is complete. Hibernate will not acquire and release connections for each lazy field loading.

Reply:
Well, if controller method returns @Entity , it is then serialized to JSON with Jackson. To generate JSON, sometimes we need to load LAZY fields. And to load them Hibernate needs to acquire a connection. And that is what I am talking about. That connection will only be released after EntityManager is closed